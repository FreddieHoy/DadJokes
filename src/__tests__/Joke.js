

import React from 'react'
import { shallow } from 'enzyme'

import FindJoke from '../components/pages/Joke'


it('should render correctly with no props', () => {
  const component = shallow(<FindJoke/>)
  expect(component).toMatchSnapshot()
})

it('should render banner text correctly with given strings', () => {
  const strings = ['one', 'two']
  const component = shallow(<FindJoke list={strings} />)
  expect(component).toMatchSnapshot()
})


it('should render button text on page', () => {
  const wrapper = shallow(<FindJoke />)
  const button = wrapper.find('button').text()

  expect(button).toEqual('prompt him for a joke')
})
