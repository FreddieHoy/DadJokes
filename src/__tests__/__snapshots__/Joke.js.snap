// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render banner text correctly with given strings 1`] = `
<div
  id="find"
>
  <h2>
    lets have it then dad.. ðŸ™„
  </h2>
  <button
    onClick={[Function]}
  >
    prompt him for a joke
  </button>
  <p>
    <i
      className="fas fa-quote-left"
    />
    <br />
    <br />
    <span />
    <br />
    <br />
    <i
      className="fas fa-quote-right"
    />
  </p>
  <h3
    onClick={[Function]}
  >
    save for later
    <span
      className={null}
    >
      <i
        className="fas fa-star"
      />
    </span>
  </h3>
</div>
`;

exports[`should render correctly with no props 1`] = `
<div
  id="find"
>
  <h2>
    lets have it then dad.. ðŸ™„
  </h2>
  <button
    onClick={[Function]}
  >
    prompt him for a joke
  </button>
  <p>
    <i
      className="fas fa-quote-left"
    />
    <br />
    <br />
    <span />
    <br />
    <br />
    <i
      className="fas fa-quote-right"
    />
  </p>
  <h3
    onClick={[Function]}
  >
    save for later
    <span
      className={null}
    >
      <i
        className="fas fa-star"
      />
    </span>
  </h3>
</div>
`;
